### ISSUE TITLE
[useToaster] - TypeError: Cannot read properties of undefined (reading '_context')

### ISSUE THREAD

**Issue by AdrienFromToulouse** [rsuite/rsuite#4033]
### What version of rsuite are you using?

5.74.0

### What version of React are you using?

18.3.1

### What version of TypeScript are you using (if any)?

5.6.3

### What browser are you using?

Chrome

### Describe the Bug

Hi there!

I would like to report an issue regarding the `useToaster` and specifically the impact that this change https://github.com/rsuite/rsuite/pull/4012 seems to have.

We are running unit tests using Jest. We test individual component, some of them use `const toaster = useToaster();`.
Since the update to `5.74.0` the following error is thrown:

```
TypeError: Cannot read properties of undefined (reading '_context')
```

I tried to fix it by wrapping the component under test with a `CustomProvider` like bellow but it does not solve the issue:

```js
 import { CustomProvider } from 'rsuite';

 .......

 test('This test fail', async () => {
    render(
      
        <SWRConfig
          value={{
            fetcher: fetcher,
            revalidateOnFocus: false,
          }}
        >
          <CustomProvider>
            <MyComponentUnderTest notifySubmit={handleSubmit} />
          </CustomProvider>
        </SWRConfig>,
    );

    expect(screen.getByText('foo bar baz.')).toBeInTheDocument();

    const deleteButton = screen.getByRole('button');
    await userEvent.click(deleteButton);

    await waitFor(() => expect(handleSubmit).toHaveBeenCalledTimes(1));
    await waitFor(() => expect(screen.queryByText('foo bar baz')).toBeInTheDocument());
  });
```

Anything I should do differently?

Cheers,

### Expected Behavior

Test should pass without issue similar to version `5.73.1`

**Comment by simonguo** [comment]
Fixed in https://github.com/rsuite/rsuite/pull/4032

```bash
npm i rsuite@5.74.1
```

**Comment by AdrienFromToulouse** [comment]
> Fixed in #4032
> 
> ```shell
> npm i rsuite@5.74.1
> ```

oh sorry I did not see the patched version, will give it a try. Thank you very much for the quick answer ❤️

**Comment by AdrienFromToulouse** [comment]
It works! 🙏
